game-core/src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1443,20 +1452,16 @@ public static Route getBestRoute(final Territory start, final Territory end, fin
     final boolean isNeutralsImpassable =
         isNeutralsImpassable(data) || (hasAir && !Properties.getNeutralFlyoverAllowed(data));
     final Predicate<Territory> noNeutral = Matches.territoryIsNeutralButNotWater().negate();
-    final Predicate<Territory> noAa = Matches.territoryHasEnemyAaForFlyOver(player, data).negate();
```

Comment:
The changes here are mostly just small refactorings to improve readability and then calling the new `getRouteIgnoreEndValidatingCanals` method.